src/afl-fuzz-run.c

```diff
@@ -488,10 +507,23 @@ void sync_fuzzers(afl_state_t *afl) {
 u8 trim_case(afl_state_t *afl, struct queue_entry *q, u8 *in_buf) {
 
   /* Custom mutator trimmer */
-  if (afl->mutator && afl->mutator->afl_custom_trim) {
+  if (afl->number_of_custom_mutators) {
+
+    u8 trimmed_case;
+    bool custom_trimmed = false;
+
+    LIST_FOREACH(&afl->custom_mutator_list, struct custom_mutator, {
+
+      if (el->afl_custom_trim) {
+        
+        trimmed_case = trim_case_custom(afl, q, in_buf, el);
```

Comment:
Not sure how often trim should be called/if they should be called all the time?

Reply:
Isn't trim called only once for each test case we have?

Reply:
I guess. Still, you call one trim for each mutator, and then only return the result of the last in case it worked?
I don't think this works for n>1 mutators, what do you think?

Reply:
Yes. I mean this and the `custom_pre_save` thing could be solved by making the first custom_mutator as "defualt" mutator? Then these heavy operations (like `trimming` and `pre_save`) could only be defined in that one mutator. What do you think?

Reply:
it returns 1 if there was trimming, right? if so it should return one if any of the custom trimmer trimmed the entry

Reply:
IMHO if a user wants different trimmers he should have them

Reply:
> it returns 1 if there was trimming, right? if so it should return one if any of the custom trimmer trimmed the entry

Yes, that is what is happening right now. :slightly_smiling_face:

Reply:
Just looked it up in the code - The return values are... odd.
However I think the current code is correct, then, the only thing is it could return earlier on errors.

Reply:
Doesn't the `trim_case_custom` function handle this? I mean it exits on any error in trimming, e,g if the new corpus size increases than before